lte/gateway/python/integ_tests/s1aptests/test_attach_detach_setsessionrules_tcp_data.py

```diff
@@ -0,0 +1,248 @@
+        ]
+        wait_for_s1 = [True, False]
+        self._s1ap_wrapper.configUEDevice(num_ues)
+        datapath = get_datapath()
+        MAX_NUM_RETRIES = 5
+
+        for i in range(num_ues):
+            req = self._s1ap_wrapper.ue_req
+            print(
+                "********************** Running End to End attach for ",
+                "UE id ",
+                req.ue_id,
+            )
+            # Now actually complete the attach
+            self._s1ap_wrapper._s1_util.attach(
+                req.ue_id,
+                s1ap_types.tfwCmd.UE_END_TO_END_ATTACH_REQUEST,
+                s1ap_types.tfwCmd.UE_ATTACH_ACCEPT_IND,
+                s1ap_types.ueAttachAccept_t,
+            )
+
+            # Wait on EMM Information from MME
+            self._s1ap_wrapper._s1_util.receive_emm_info()
+
+            # UL Flow description #1
+            ulFlow1 = {
+                "ip_proto": "TCP",  # Protocol Type
+                "direction": "UL",  # Direction
+            }
+
+            # DL Flow description #1
+            dlFlow1 = {
+                "ip_proto": "TCP",  # Protocol Type
+                "direction": "DL",  # Direction
+            }
+
+            # Flow list to be configured
+            flow_list = [
+                ulFlow1,
+                dlFlow1,
+            ]
+
+            # QoS
+            qos = {
+                "qci": 8,  # qci value [1 to 9]
+                "priority": 0,  # Range [0-255]
+                "max_req_bw_ul": 10000000,  # MAX bw Uplink
+                "max_req_bw_dl": 15000000,  # MAX bw Downlink
+                "gbr_ul": 1000000,  # GBR Uplink
+                "gbr_dl": 2000000,  # GBR Downlink
+                "arp_prio": 15,  # ARP priority
+                "pre_cap": 1,  # pre-emption capability
+                "pre_vul": 1,  # pre-emption vulnerability
+            }
+
+            policy_id = "tcp"
+
+            print("Sleeping for 5 seconds")
+            time.sleep(5)
+            print(
+                "********************** Set Session Rule for IMSI",
+                "".join([str(i) for i in req.imsi]),
+            )
+            self._sessionManager_util.send_SetSessionRules(
+                "IMSI" + "".join([str(i) for i in req.imsi]),
+                policy_id,
+                flow_list,
+                qos,
+            )
+
+            # Receive Activate dedicated bearer request
+            response = self._s1ap_wrapper.s1_util.get_response()
+            self.assertEqual(
+                response.msg_type, s1ap_types.tfwCmd.UE_ACT_DED_BER_REQ.value
+            )
+            act_ded_ber_ctxt_req = response.cast(
+                s1ap_types.UeActDedBearCtxtReq_t
+            )
+
+            # Send Activate dedicated bearer accept
+            self._s1ap_wrapper.sendActDedicatedBearerAccept(
+                req.ue_id, act_ded_ber_ctxt_req.bearerId
+            )
+
+            policy_id = "tcp2"
```

Comment:
Why is the same policy getting added twice?

Reply:
The purpose is to see if the first policy is overwritten by the second policy. This triggers a delete bearer and create bearer requests. The test will need to check enforcement rules (whether they are overwritten or not). I plan to change the requested rates and verify the ul/dl speeds as well.